src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -69,74 +78,78 @@ public class SecretManager {
     private final AWSSecretClient secretClient;
     private final SecretStore<SecretDocument, AWSSecretResponse> secretStore;
     @Nullable
-    private Crypter crypter;
-    private final Result<SecretCryptoException> initialized = new Result<>();
+    private final AtomicReference<Crypter> crypter = new AtomicReference<>();
 
     /**
      * Constructor.
-     * @param secretClient client for aws secrets.
+     *
+     * @param secretClient    client for aws secrets.
      * @param securityService security service.
-     * @param dao          dao for persistent store.
+     * @param dao             dao for persistent store.
      */
     @Inject
     SecretManager(AWSSecretClient secretClient, SecurityService securityService, FileSecretStore dao,
-                  ExecutorService executor) {
+                  DeviceConfiguration deviceConfiguration) {
         this.secretStore = dao;
         this.secretClient = secretClient;
+        this.securityService = securityService;
+        deviceConfiguration.onAnyChange(((whatHappened, node) -> {
+            if (validUpdate(node, DEVICE_PARAM_CERTIFICATE_FILE_PATH) || validUpdate(node,
+                    DEVICE_PARAM_PRIVATE_KEY_PATH)) {
+                // TODO: If the device creds change, then cache and local store should be updated as well as the keys
+                //  used for encrypting and decrypting the secrets have changed.
+                synchronized (crypter) {
+                    crypter.set(null);
+                }
+            }
+        }));
+    }
+
+    private boolean validUpdate(Node node, String key) {
+        return node != null && node.childOf(key) && Utils.isNotEmpty(Coerce.toString(node)) && Files.exists(
+                Paths.get(key));
```

Comment:
You cannot check that a path exists on the filesystem. We support non-file based crypto